ui/src/app/components/dataflow-form/dataflow-form.component.ts

```diff
@@ -22,8 +24,19 @@ export class DataflowFormComponent implements OnInit {
       numWorkers: new FormControl('1',[Validators.required, Validators.pattern('^[1-9][0-9]*$')]),
       maxWorkers: new FormControl('50', [Validators.required, Validators.pattern('^[1-9][0-9]*$')]),
       serviceAccountEmail: new FormControl(''),
-      hostProjectId: new FormControl(data.GCPProjectID, Validators.required),
+      vpcHostProjectId: new FormControl(data.GCPProjectID, Validators.required),
+      machineType: new FormControl(''),
+      // With the json validator, the form still allows nested json values, which is not a valid entry for user labels.
+      // Wwe rely on the Dataflow validation for nested json validation.
+      additionalUserLabels: new FormControl('', [this.jsonValidator]),
```

Comment:
Why can't we write a custom validator?

Reply:
NA since found the regex for additional user labels.